runtime/android/core/src/org/xwalk/core/XWalkCoreWrapper.java

```diff
@@ -150,14 +157,19 @@ public static int attachXWalkCore(Context context) {
         Log.d(TAG, "Attach xwalk core");
         sProvisionalInstance = new XWalkCoreWrapper(context, -1);
         if (!sProvisionalInstance.findEmbeddedCore()) {
-            int status = sProvisionalInstance.mCoreStatus;
+            if(sProvisionalInstance.isDownloadMode()) {
```

Comment:
What if the check fails in embedded/download modes but success in shared mode and the user actually wanted to use their specified XWalkRuntimeLib.apk downloaded from their hosts?